embedmd/content.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to writing, software distributed
+// under the License is distributed on a "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package embedmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+)
+
+// Fetcher provides an abstraction on a file system.
+// The Fetch function is called anytime some content needs to be fetched.
+// For now this includes files and URLs.
+type Fetcher interface {
+	Fetch(dir, path string) ([]byte, error)
```

Comment:
It would be nice to document `dir` and `path`. How are they used? Why are there 2 parameters instead of just one (`path`)? It's not very obvious to me based on the interface description alone.